modules/survey_accounts/test/survey_accountsTest.php

```diff
@@ -39,5 +147,110 @@ function testSurveyAccountsAddSurveyDoespageLoad()
+           )->sendKeys("8889");
+          $this->safeFindElement(
+               WebDriverBy::Name("fire_away")
+           )->click();
+           $bodyText =  $this->safeFindElement(
+               WebDriverBy::cssSelector(".error")
+           )->getText();
+           $this->assertContains("Visit does not exist for given candidate", $bodyText);
+          //PSCID and DCC ID do not match or candidate does not exist.
+          $this->safeFindElement(
+               WebDriverBy::Name("CandID")
+           )->sendKeys("888888");
+          $this->safeFindElement(
+               WebDriverBy::Name("PSCID")
+           )->sendKeys("8889");
+          $this->safeFindElement(
+               WebDriverBy::Name("fire_away")
+           )->click();
+           $bodyText =  $this->safeFindElement(
+               WebDriverBy::cssSelector(".error")
+           )->getText();
+           $this->assertContains("PSCID and DCC ID do not match or candidate does not exist", $bodyText);
+    }
+
+   /**
+     * Tests clear button in the filter section, input some data, then click the clear button,
+     * all of data in the filter section will be gone. 
+     *
+     * @return void
+     */
+    function testSurveyAccountsClearButton()
+    {
+        //testing the PSCID
+        $this->safeGet($this->url . "/survey_accounts/");
+        $this->webDriver->findElement(WebDriverBy::Name("PSCID"))->sendKeys("test");
+        $this->webDriver->findElement(WebDriverBy::Name("reset"))->click();
+        $bodyText = $this->webDriver->findElement(WebDriverBy::Name("PSCID"))
+               ->getText();
+        $this->assertEquals("", $bodyText);
+
+        //testing the Email
+        $this->webDriver->findElement(WebDriverBy::Name("Email"))->sendKeys("test");
+        $this->webDriver->findElement(WebDriverBy::Name("reset"))->click();
+        $bodyText = $this->webDriver->findElement(WebDriverBy::Name("Email"))
+               ->getText();
+        $this->assertEquals("", $bodyText);
+    }
+   /**
+     * Tests that, input some data and click search button, check the results.
+     *
+     * @return void
+     */
+    function testSurveyAccountsSearchButton()
+    {
+        //testing search by PSCID
+        $this->safeGet($this->url . "/survey_accounts/");
+        $this->webDriver->findElement(WebDriverBy::Name("PSCID"))->sendKeys
+             ("8888");
+        $this->webDriver->findElement(WebDriverBy::Name("filter"))->click();
+        $bodyText = $this->webDriver->getPageSource();
+        $this->assertContains("8888", $bodyText);
+
+        //testing search by Email
+        $this->safeGet($this->url . "/survey_accounts/");
+        $this->webDriver->findElement(WebDriverBy::Name("Email"))
+             ->sendKeys("TestTestTest@gmail.com");
+        $this->webDriver->findElement(WebDriverBy::Name("filter"))->click();
+        $bodyText = $this->webDriver->getPageSource();
+        $this->assertContains("TestTestTest@gmail.com", $bodyText);
```

Comment:
I think there might be a problem with this test not testing properly. The insert above in the setup is `@example.com`, but the search is `@gmail.com`, so this assertion should fail, but isn't. I think it's finding the text inside of the input box where you do user input, not inside of the results. This probably needs a smarter xpath selection/assertion.